### PR TITLE
feat(status): reimpl status page

### PR DESCRIPTION
**Pull request by Yixuan-Wang**
Add status page, reimplementing [the original page](http://mirrors.pku.edu.cn/Status).

### CODE REVIEW

**Comment by Yixuan-Wang** [comment]
for a quick glance:

image

File: composables/status.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import type { EChartsOption } from 'echarts'
```
Comment (thezzisu) [inline composables/status.ts]: This is not a composable. Please move it into /lib/ like the main branch.

File: package.json [hunk 2]
```diff
@@ -29,5 +29,9 @@
     "prettier": "^2.8.4",
     "typescript": "^4.9.5",
     "vue-tsc": "^1.2.0"
+  },
```
Comment (thezzisu) [inline package.json]: It's recommended to move pre-build dependencies into devDependencies.

File: plugins/chart.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { LineChart } from 'echarts/charts'
```
Comment (thezzisu) [inline plugins/chart.ts]: Nuxt plugin will be applied to every page, which will cause other pages to slow down. Import these components in status page instead.

File: components/StatusBase.vue [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<template>
```
Comment (thezzisu) [inline components/StatusBase.vue]: You could merge Status*.vue into one component.
Only formatter and URL are different, use props to pass them.

**Review by thezzisu — CHANGES_REQUESTED** [review]
Requested changes are stated in per-file comments. Nice work, fix and merge!

**Review by thezzisu — APPROVED** [review]
Good job.

**Pull request merged**